src/panels/config/integrations/integration-panels/zwave/zwave-migration.ts

```diff
@@ -0,0 +1,480 @@
+                        migrated you might have to change those manually.
+                      </div>
+                      <div class="card-actions">
+                        <mwc-button @click=${this._navigateOzw}>
+                          Go to OZW config panel
+                        </mwc-button>
+                      </div>
+                    </ha-card>`
+                  : ""}
+              `}
+        </ha-config-section>
+      </hass-subpage>
+    `;
+  }
+
+  private async _getMigrationConfig(): Promise<void> {
+    this._migrationConfig = await fetchMigrationConfig(this.hass!);
+  }
+
+  private async _unsubscribe(): Promise<void> {
+    if (this._unsub) {
+      (await this._unsub)();
+      this._unsub = undefined;
+    }
+  }
+
+  private _continue(): void {
+    this._step++;
+  }
+
+  private async _stopNetwork(): Promise<void> {
+    this._stoppingNetwork = true;
+    await this._getNetworkStatus();
+    if (this._networkStatus?.state === ZWAVE_NETWORK_STATE_STOPPED) {
+      this._networkStopped();
+      return;
+    }
+
+    this._unsub = this.hass!.connection.subscribeEvents(
+      () => this._networkStopped(),
+      "zwave.network_stop"
+    );
+    this.hass!.callService("zwave", "stop_network");
+  }
+
+  private async _setupOzw() {
+    const ozwConfigFlow = await startOzwConfigFlow(this.hass);
+    if (
+      !this.hass.config.components.includes("hassio") &&
+      this.hass.config.components.includes("ozw")
+    ) {
+      this._getMigrationData();
+      this._step = 3;
+      return;
+    }
+    showConfigFlowDialog(this, {
+      continueFlowId: ozwConfigFlow.flow_id,
+      dialogClosedCallback: () => {
+        if (this.hass.config.components.includes("ozw")) {
+          this._getMigrationData();
+          this._step = 3;
+        }
+      },
+      showAdvanced: this.hass.userData?.showAdvanced,
+    });
+    this.hass.loadBackendTranslation("title", "ozw", true);
+  }
+
+  private async _getMigrationData() {
+    this._migrationData = await migrateZwave(this.hass, true);
+    this._migratedZwaveEntities = Object.keys(
+      this._migrationData.migration_entity_map
+    );
+    if (Object.keys(this._migrationData.migration_device_map).length) {
+      this._deviceRegistry = await fetchDeviceRegistry(this.hass);
+    }
+  }
+
+  private _computeDeviceName(deviceId) {
+    const device = this._deviceRegistry?.find(
+      (devReg) => devReg.id === deviceId
+    );
+    if (!device) {
+      return deviceId;
+    }
+    return computeDeviceName(device, this.hass);
+  }
+
+  private async _doMigrate() {
+    const data = await migrateZwave(this.hass, false);
+    if (!data.migrated) {
+      showAlertDialog(this, { title: "Migration failed!" });
+      return;
+    }
+    this._step = 4;
+  }
+
+  private _navigateOzw() {
+    navigate(this, "/config/ozw");
+  }
+
+  private _networkStopped(): void {
+    this._unsubscribe();
+    this._getMigrationConfig();
```

Comment:
Why is this not with async/await?

Reply:
Because they are unrelated to each other, we don't have to wait for the first to finish to start the second.